dandi/pynwb_utils.py

```diff
@@ -227,9 +231,95 @@ def _get_pynwb_metadata(path):
             key = f[len("number_of_") :]
             out[f] = len(getattr(nwb, key, []) or [])
 
+        # get external_file data:
+        out["external_file_objects"] = _get_image_series(nwb)
+
+    return out
+
+
+def _get_image_series(nwb: pynwb.NWBFile) -> List[dict]:
+    """
+    This method supports _get_pynwb_metadata() in retrieving all ImageSeries
+    related metadata from an open nwb file.
+    Specifically it pulls out the ImageSeries uuid, name and all the
+    externally linked files
+    named under the argument 'external_file'.
+    Parameters
+    ----------
+    nwb: pynwb.NWBFile
+    Returns
+    -------
+    out: List[dict]
+        list of dicts : [{id: <ImageSeries uuid>, name: <IMageSeries name>,
+        external_files=[ImageSeries.external_file]}]
+        if no ImageSeries found in the given modules to check, then it returns an empty list.
+    """
+    out = []
+    for module_name in external_file_modules:
+        module_cont = getattr(nwb, module_name)
+        for name, ob in module_cont.items():
+            if isinstance(ob, pynwb.image.ImageSeries) and ob.external_file is not None:
+                out_dict = dict(id=ob.object_id, name=ob.name, external_files=[])
+                for ext_file in ob.external_file:
+                    if Path(ext_file).suffix in external_file_extensions:
+                        out_dict["external_files"].append(Path(ext_file))
+                    else:
+                        lgr.warning(
+                            "external file %s should be one of: %s",
+                            ext_file,
+                            external_file_extensions,
+                        )
+                out.append(out_dict)
```

Comment:
could there be some sample file(s) created (or worse to worse - shipped along)  to be able to design a test which would cover this notable in size codeblock?

Reply:
probably just disregard this comment for now since it seems work needs to be done to address failing tests first, and then coverage would get uploaded possibly invalidating my comment above ;)